### PR TITLE
use monaco editor in terminal accessible buffer

### LINKED ISSUE #174078: Selection doesn't work in the content editable accessible buffer
1. Set `terminal.integrated.accessibleBufferContentEditable: on`
2. focus a terminal
3. run some commands
4. `Shift+Tab` and try to select 🐛 

fyi @Tyriar

### PR DESCRIPTION
**Pull request by meganrogge**
fix https://github.com/microsoft/vscode/issues/174078

Part of #174079

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts [hunk 1]
```diff
@@ -763,76 +771,119 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
...
-		this._accessibleBuffer.focus();
-		if (this._accessibleBuffer.contentEditable === 'true') {
-			document.execCommand('selectAll', false, undefined);
-			document.getSelection()?.collapseToEnd();
-		} else if (header) {
-			// focus the cursor line's header
-			header.tabIndex = 0;
-		}
-		return this._bufferElementFragment;
+		return content;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:887]: It'll save a bunch of memory being allocated if you do the `lines: string[] = []`/`lines.join('\n')` approach (or stringbuilder again?)

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts [hunk 2]
```diff
@@ -763,76 +771,119 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
...
-		for (const command of commands) {
-			header = document.createElement('h2');
-			// without this, the text area gets focused when keyboard shortcuts are used
-			header.tabIndex = -1;
-			header.textContent = command.command.replace(new RegExp(' ', 'g'), '\xA0');
-			if (command.exitCode !== 0) {
-				header.textContent += ` exited with code ${command.exitCode}`;
+		const sb = new StringBuilder(10000);
+		let content = localize('terminal.integrated.noContent', "No terminal content available for this session. Run some commands to create content.");
+		if (commands?.length) {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:857]: An early exit would be good here:

```ts
if (!commands?.length) {
  return localize('terminal.integrated.noContent', "No terminal content available for this session. Run some commands to create content.");
}
```

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts [hunk 3]
```diff
@@ -281,7 +289,7 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
 		});
 	}
 
-	focusAccessibleBuffer(): void {
+	async focusAccessibleBuffer(): Promise<void> {
 		this._accessibileBuffer?.focus();
 	}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:294]: Was this an accident? Doesn't need to be async

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts [hunk 4]
```diff
@@ -763,76 +771,119 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
 		this.raw.write(data);
 	}
 }
-
+const ACCESSIBLE_BUFFER_SCHEME = 'terminal-accessible-buffer';
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:774]: const enum is preferred so it's inlined

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts [hunk 5]
```diff
@@ -763,76 +771,119 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
...
-	private _bufferElementFragment: DocumentFragment | undefined;
+	private _accessibleBuffer: HTMLElement;
+	private _bufferEditor: CodeEditorWidget;
+	private _editorContainer: HTMLElement;
+	private _registered: boolean = false;
 
 	constructor(
 		private readonly _terminal: RawXtermTerminal,
-		private readonly _font: ITerminalFont,
+		font: ITerminalFont,
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:783]: Font can change, do we need to listen to config service?

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts (hunk 3 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts:294]: It wasn't an accident

```ts
const model = await this._getTextModel(URI.from({ scheme: ACCESSIBLE_BUFFER_SCHEME, fragment }));
```

**Review by lramos15 — APPROVED** [review]

**Pull request merged**